openpype/modules/deadline/abstract_submit_deadline.py

```diff
@@ -66,6 +67,58 @@ def requests_get(*args, **kwargs):
     return requests.get(*args, **kwargs)
 
 
+class DeadlineIndexedVar(dict):
+    """
+
+    Allows to set and query values by integer indices:
+        Query: var[1] or var.get(1)
+        Set: var[1] = "my_value"
+        Append: var += "value"
+
+    Note: Iterating the instance is not guarantueed to be the order of the
+          indices. To do so iterate with `sorted()`
+
+    """
+    def __init__(self, key):
+        self.__key = key
+
+    def next_available_index(self):
+        # Add as first unused entry
+        i = 0
+        while i in self.keys():
+            i += 1
+        return i
+
+    def serialize(self):
+        key = self.__key
+
+        # Allow custom location for index in serialized string
+        if "{}" not in key:
+            key = key + "{}"
+
+        return {
+            key.format(index): value for index, value in sorted(self.items())
+        }
+
+    def update(self, data):
+        # Force the integer key check
+        for key, value in data.items():
+            self.__setitem__(key, value)
+
+    def __iadd__(self, other):
+        index = self.next_available_index()
+        self[index] = other
+        return self
+
+    def __setitem__(self, key, value):
+        if not isinstance(key, int):
+            raise TypeError("Key must be an integer: {}".format(key))
+
+        if key < 0:
+            raise ValueError("Negative index can't be set: {}".format(key))
+        dict.__setitem__(self, key, value)
```

Comment:
Likely a better implementation can be build for this, but I wanted to make it easy to set specific indices whilst also allowing the ability to easily append values like was done before with the old setters/getters.
```python
# Append value
job_info.EnvironmentKeyValue += "key=value"
# Query index (must exist)
job_info.EnvironmentKeyValue[0]
# Get index
job_info.EnvironmentKeyValue.get(0)
# Set by index
job_info.EnvironmentKeyValue[0] = "hello"
```

Setting/querying specific indices might not be super relevant for `EnvironmentKeyValue` but it is instead for `ExtraExtraInfoKeyValue`.

I'd have loved if the implementation would also allow to "uniqify" added keys (e.g. for `EnvironmentKeyValue` or `AssetDependency`) but that isn't super crucial.